pkg/util/storage.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"crypto/x509"
+	"fmt"
+	"net/url"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/operator-framework/rukpak/internal/storage"
+	"github.com/operator-framework/rukpak/internal/util"
+	pkgstorage "github.com/operator-framework/rukpak/pkg/storage"
+)
+
+const DefaultBundleCacheDir = "/var/cache/bundles"
+
+func NewBundleStorage(mgr ctrl.Manager, provisionerStorageDirectory, httpExternalAddr, bundleCAFile, bearerToken string) (pkgstorage.Storage, error) {
```

Comment:
A few more improvements I'd suggest:
- Use the functional options pattern for bundleCAFile and bearerToken
- For bundleCAFile, accept an `io.Reader` instead of a filename. That will make it easier to adapt this to other use cases (e.g. unit testing)

Reply:
Heavy +1 on the io.Reader suggestion here.

Reply:
I updated the bundleCAFile to `io.Reader" and to use the functional options pattern.